pandas/core/arrays/categorical.py

```diff
@@ -1374,19 +1374,15 @@ def memory_usage(self, deep=False):
     @Substitution(klass='Categorical')
     @Appender(_shared_docs['searchsorted'])
     def searchsorted(self, value, side='left', sorter=None):
-        if not self.ordered:
-            raise ValueError("Categorical not ordered\nyou can use "
-                             ".as_ordered() to change the Categorical to an "
-                             "ordered one")
-
         from pandas.core.series import Series
-        codes = _get_codes_for_values(Series(value).values, self.categories)
-        if -1 in codes:
+        value_codes = _get_codes_for_values(Series(value).values,
+                                            self.categories)
+        if -1 in value_codes:
             raise KeyError("Value(s) to be inserted must be in categories.")
+        value_codes = value_codes[0] if is_scalar(value) else value_codes
 
-        codes = codes[0] if is_scalar(value) else codes
-
-        return self.codes.searchsorted(codes, side=side, sorter=sorter)
+        return algorithms.searchsorted(self._codes, value_codes,
```

Comment:
we don't have a guarantee that these codes are sorted right? so this seems likely to return an incorrect result if that is not the case

Reply:
I've added tests for this ``test_searchsorted_unordered``.